### DISCUSSION TITLE
https://greendolphindance.com/2024/02/16/%E5%88%B6%E4%BD%9Chexo%E6%97%B6%E9%97%B4%E8%BD%B4/

### DISCUSSION THREAD

**Discussion by giscus** [greendolphindance/greendolphindance.github.io#17] [General]
# https://greendolphindance.com/2024/02/16/%E5%88%B6%E4%BD%9Chexo%E6%97%B6%E9%97%B4%E8%BD%B4/

Back to the rough ground!

https://greendolphindance.com/2024/02/16/%E5%88%B6%E4%BD%9Chexo%E6%97%B6%E9%97%B4%E8%BD%B4/

**Comment by wang7peng** [comment]
谢谢学霸大妹妹踩坑，然鹅我没有成功 :cry: 。`<div class="timeline">` 下面这行里调用了个 `getYaml` 方法，我在 `hexo g` 的时候提示 **ERROR** `getYaml is not defined` 。然后最后这个 "设定样式" 小节的内容，就是直接把这些拷贝到上面 ejs 文件的 <head> 标签里就行了吧？

**Reply by greendolphindance** [reply]
诶！我回去看了半天，也没找到我的getYaml是在哪里定义的（好像没有定义但我的能跑？）。问了一下GPT，要不你试试js-yaml库呢？然后“设定样式”的内容可以写在ejs里，用<style>标签框起来，也可以放在外部的CSS文档里

**Reply by wang7peng** [reply]
搞定了！这是个 `php` 函数，这个文件是 `html` 代码和 `php` 代码的混合编程。就是主体是 `html` 标签，在 `<% %>` 之间加上调用的 php 代码。所以 **解决方案** 是 `npm install -g node-php`   🥴

**Reply by greendolphindance** [reply]
原来如此！
